src/transactions/OfferExchange.cpp

```diff
@@ -1248,6 +1248,19 @@ exchangeWithPool(int64_t reservesToPool, int64_t maxSendToPool, int64_t& toPool,
     {
     case RoundingType::PATH_PAYMENT_STRICT_SEND:
     {
+        // PathPaymentStrictSend always allows INT64_MAX to be received at
+        // every hop, and exchange with a pool always happens as a unit
+        // (compare against the order book where multiple offers might execute
+        // in a hop).
+        if (maxReceiveFromPool != INT64_MAX)
```

Comment:
Having a check like this (esp. given line 1260) seems like the argument for having two different `exchangeWithPool()` variants (depending on strict send or strict recv), but this isn't my code-base heh.